edb/edgeql/codegen.py

```diff
@@ -1808,19 +1817,82 @@ def _after_index(self, node: qlast.IndexCommand) -> None:
             self.visit(node.except_expr)
             self.write(')')
 
+    def visit_IndexType(
+        self,
+        node: qlast.IndexType
+    ) -> None:
+        if isinstance(node.name, tuple):
+            self.write(
+                f'{ident_to_str(node.name[0])}::{ident_to_str(node.name[1])}')
+        else:
+            self.write(ident_to_str(node.name))
+
+        if node.kwargs:
+            self.write('(')
+            for i, (name, arg) in enumerate(node.kwargs.items()):
+                if i > 0:
+                    self.write(', ')
+                self.write(f'{edgeql_quote.quote_ident(name)} := ')
+                self.visit(arg)
+            self.write(')')
+
     def visit_CreateIndex(self, node: qlast.CreateIndex) -> None:
+        def after_name() -> None:
+            if node.params:
```

Comment:
What are params, here? How do they interact with kwargs?

Reply:
Oh, I understand now. Only one can be present, though, right? Maybe have an assert to that effect?

Reply:
No args or params in this iteration of the PR anymore.